connectors/tests/test_byoc.py

```diff
@@ -959,12 +960,44 @@ def test_nested_get(nested_dict, keys, default, expected):
     assert expected == Features(nested_dict)._nested_feature_enabled(keys, default)
 
 
-JOB_SOURCE = {"_id": "1", "_source": {"status": "pending", "connector": {"id": "1"}}}
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "sync_now, trigger_method",
+    [
+        (True, JobTriggerMethod.ON_DEMAND),
+        (False, JobTriggerMethod.SCHEDULED),
+    ],
+)
+@patch("connectors.byoc.SyncJobIndex.index")
+async def test_create_job(
+    index_method, sync_now, trigger_method, patch_logger, set_env
```

Comment:
```suggestion
    index_method, sync_now, trigger_method, set_env
```

I think we only need `set_env` to set `elasticsearch.password` in the os environment. `patch_logger` seems unnecessary.

Reply:
`patch_logger` is to silent logs.

compare this (with `patch_logger`):
```
chenhui.wang@SG-Chenhui connectors-python % bin/pytest -sv connectors/tests/test_byoc.py -k test_create_job
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 3.10.7, pytest-7.1.3, pluggy-1.0.0 -- /Users/chenhui.wang/src/github.com/elastic/connectors-python/bin/python3.10
cachedir: .pytest_cache
Using --randomly-seed=2671744692
rootdir: /Users/chenhui.wang/src/github.com/elastic/connectors-python
plugins: fail-slow-0.3.0, asyncio-0.19.0, randomly-3.12.0, cov-3.0.0
asyncio: mode=strict
collected 69 items / 67 deselected / 2 selected                                                                                                                                                                                                                              

connectors/tests/test_byoc.py::test_create_job[True-JobTriggerMethod.ON_DEMAND] PASSED
connectors/tests/test_byoc.py::test_create_job[False-JobTriggerMethod.SCHEDULED] PASSED

====================================================================================================================== 2 passed, 67 deselected in 0.24s ======================================================================================================================
```

with this (without `patch_logger`):
```
chenhui.wang@SG-Chenhui connectors-python % bin/pytest -sv connectors/tests/test_byoc.py -k test_create_job
============================================================================================================================ test session starts =============================================================================================================================
platform darwin -- Python 3.10.7, pytest-7.1.3, pluggy-1.0.0 -- /Users/chenhui.wang/src/github.com/elastic/connectors-python/bin/python3.10
cachedir: .pytest_cache
Using --randomly-seed=147054654
rootdir: /Users/chenhui.wang/src/github.com/elastic/connectors-python
plugins: fail-slow-0.3.0, asyncio-0.19.0, randomly-3.12.0, cov-3.0.0
asyncio: mode=strict
collected 69 items / 67 deselected / 2 selected                                                                                                                                                                                                                              

connectors/tests/test_byoc.py::test_create_job[False-JobTriggerMethod.SCHEDULED] [FMWK][17:57:24][INFO] Loading config from /Users/chenhui.wang/src/github.com/elastic/connectors-python/connectors/tests/config.yml
PASSED
connectors/tests/test_byoc.py::test_create_job[True-JobTriggerMethod.ON_DEMAND] [FMWK][17:57:24][INFO] Loading config from /Users/chenhui.wang/src/github.com/elastic/connectors-python/connectors/tests/config.yml
PASSED

====================================================================================================================== 2 passed, 67 deselected in 0.23s ======================================================================================================================
```

Reply:
Ok, I see, thank you for explaining! Maybe it's worth to have a separate fixture `silent_logger` / `deactivate_logging`, which is only there for deactivating logging. It's not obvious that `patch_logger` has this side effect. Could be done in a separate PR.

Reply:
`patch_logger` is implemented not to silence logs, but to be able to assert on logs. I don't think silent logs are good - they are instead making it much much harder to debug a problem once the test fails.

Reply:
I'd love to change it to False eventually if nobody is against it